pkg/depth/buffer.go

```diff
@@ -163,29 +175,48 @@ func (b *Buffer) AddUpdate(o types.SliceOrderBook, firstUpdateID int64, finalArg
 
 	// skip older events
 	if u.FinalUpdateID <= b.finalUpdateID {
-		b.logger.Infof("the final update id %d of event is less than equal to the final update id %d of the snapshot, skip", u.FinalUpdateID, b.finalUpdateID)
+		b.logger.Infof("the final update id %d of event is less than equal to the final update id %d of the snapshot, skip",
+			u.FinalUpdateID, b.finalUpdateID)
 		b.mu.Unlock()
 		return nil
 	}
 
 	// if there is a missing update, we should reset the snapshot and re-fetch the snapshot
-	if u.FirstUpdateID > b.finalUpdateID+1 {
-		// drop the prior updates in the buffer since it's corrupted
-		b.buffer = []Update{u}
-		b.resetSnapshot()
-		b.once.Reset()
-		b.once.Do(func() {
-			b.logger.Info("try fetching the snapshot due to missing update")
-			go b.tryFetch()
-		})
+	if b.isFutures {
+		previousUpdateID := b.finalUpdateID
```

Comment:
could you please document how do we need previousUpdateID for futures ? thanks!

Reply:
ok